main.js

```diff
@@ -0,0 +1,45 @@
+const toDoList = [];
+const taskInput = document.querySelector(".add input");
+const taskCounter = document.querySelector("h1 span");
+const ul = document.querySelector("ul");
+
+const renderList = () => {
+  ul.textContent = "";
+  toDoList.forEach((listElement, key) => {
+    listElement.dataset.key = key;
+    ul.appendChild(listElement);
+  });
+};
+
+const removeTask = (e) => {
+  const taskIndex = e.target.parentNode.dataset.key;
+  toDoList.splice(taskIndex, 1);
+  taskCounter.textContent = toDoList.length;
+  renderList();
+};
+
+const addTask = (e) => {
+  e.preventDefault();
+  if (taskInput.value === "") return;
+  const newTask = document.createElement("li");
+  newTask.className = "task";
+  newTask.innerHTML = taskInput.value + "<button>Usuń</button>";
+  taskInput.value = "";
+  toDoList.push(newTask);
+  taskCounter.textContent = toDoList.length;
+  ul.appendChild(newTask);
+  renderList();
+  newTask.addEventListener("click", removeTask);
+};
+
+const searchTask = (e) => {
+  const searchText = e.target.value.toLowerCase();
+  let filtr = toDoList.filter((li) =>
+    li.textContent.toLowerCase().includes(searchText)
+  );
+  ul.textContent = "";
```

Comment:
Druga sprawa, że w tym miejscu też używasz czyszczenia listy, ale tak naprawdę myśląc o interfejsie w momencie kiedy wyszukujesz w taskach chcesz wyświetlić te zawierające frazę, a pozostałe **ukryć** więc myśląc przyszłościowo, można by powiedzieć, że lepiej będzie tu pasować logika w której zamiast usuwać elementy i je re-renderować będziesz nadawać tym niechcianym jakąś klasę `hidden`

Reply:
Ale to tylko taka sugestia ofc, bo to co jest działa bdb